specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-04-01-preview/ContentTemplates.json

```diff
@@ -0,0 +1,417 @@
+        },
+        "providers": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataProviders",
+          "description": "Providers for the content item"
+        },
+        "firstPublishDate": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataFirstPublishDate",
+          "description": "first publish date content item"
+        },
+        "lastPublishDate": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataLastPublishDate",
+          "description": "last publish date for the content item"
+        },
+        "customVersion": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataCustomVersion",
+          "description": "The custom version of the content. A optional free text"
+        },
+        "contentSchemaVersion": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataContentSchemaVersion",
+          "description": "Schema version of the content. Can be used to distinguish between different flow based on the schema version"
+        },
+        "icon": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataIcon",
+          "description": "the icon identifier. this id can later be fetched from the content metadata"
+        },
+        "threatAnalysisTactics": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataThreatAnalysisTactics",
+          "description": "the tactics the resource covers"
+        },
+        "threatAnalysisTechniques": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataThreatAnalysisTechniques",
+          "description": "the techniques the resource covers, these have to be aligned with the tactics being used"
+        },
+        "previewImages": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataPreviewImages",
+          "description": "preview image file names. These will be taken from the solution artifacts"
+        },
+        "previewImagesDark": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataPreviewImagesDark",
+          "description": "preview image file names. These will be taken from the solution artifacts. used for dark theme support"
+        },
+        "packageId": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataContentId",
+          "description": "the package Id contains this template"
+        },
+        "packageKind": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataPackageKind",
+          "description": "the packageKind of the package contains this template"
+        },
+        "packageName": {
+          "$ref": "./common/ContentCommonTypes.json#/definitions/metadataDisplayName",
+          "description": "the name of the package contains this template"
+        },
+        "mainTemplate": {
+          "description": "The JSON string of the template",
+          "type": "object"
```

Comment:
again using freeform object like this is not allowed. 

Can you please explain your scenario a bit ? How are you using the template here ? Can the user provide any template they want ?

Reply:
Please address this comment.

Reply:
This is the same as the packagedContent. The mainTemplate would be the ARM templates and it varies from case to case. If there is any alternative or suggestion, please let me know. I can follow up.